src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -4669,6 +4669,33 @@ static void Main()
             CompileAndVerify(source, expectedOutput: "1").VerifyDiagnostics();
         }
 
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/69093")]
+        public void OverloadResolution_53()
+        {
+            var source = """
+                using System;
+
+                class A
+                {
+                    static void Main()
+                    {
+                        object x = 1;
+                        var a = new A();
+                        a.F1(x, y => F2(int () => y));
+                    }
+
+                    void F1<T>(T x, Func<T, int> f) => Console.Write("A");
+                    static int F2(Func<int> f) => f();
+                }
+
+                static class B
+                {
+                    public static void F1(this A a, object x, Func<int, int> f) => Console.Write("B");
+                }
+                """;
+            CompileAndVerify(source, expectedOutput: "B").VerifyDiagnostics();
```

Comment:
Yes, this behavior looks correct to me. (I realized my initial look at this case was incomplete.)

Reply:
Also, should we add a new EndToEnd test which shows that the change results in "viable" perf in the relevant scenario?

Reply:
> Also, should we add a new EndToEnd test which shows that the change results in "viable" perf in the relevant scenario?

Good suggestion, made me realize I'm not sure how skipping lambda binding is actually supposed to help - seems that the bound lambda body is cached, so normal compilations won't be sped up by skipping the binding when checking the conversion since the binding would be performed anyway later.

Reply:
> Also, should we add a new EndToEnd test which shows that the change results in "viable" perf in the relevant scenario?

Thanks, trying to write such test made me realize this might not be a complete fix. There's another place where lambdas are bound much more frequently when type inference is involved (for error recovery):

https://github.com/dotnet/roslyn/blob/579fc28bbf513db0d963ac04145236eca66fbd8e/src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs#L1765

Hence the change currently in this PR doesn't seem to improve anything in my testing. Will investigate more.

Reply:
> Also, should we add a new EndToEnd test which shows that the change results in "viable" perf in the relevant scenario?

https://github.com/dotnet/roslyn/pull/59733 includes a change to track the number of unbound lambdas and number of times lambda expressions are bound. Would something similar be useful here for measuring the impact?